lib/scripts/controllers/popup.js

```diff
@@ -32,14 +34,53 @@ export class PopupCtrl {
     if (this.activation !== value) {
       this.activation = value;
       chrome.storage.local.set({activation: value});
-      chrome.tabs.getSelected(null, function(tab) {
+      chrome.tabs.getSelected(null, (tab) => {
+        this.autoBlacklistWebsite(tab.url);
         var code = 'window.location.reload();';
         chrome.tabs.executeScript(tab.id, {code: code});
       });
     }
   }
 
   /**
+   * Given a url, it checks whether a URL should be
+   * blacklisted or not.
+   * @param {string} - website URL
+   */
+  autoBlacklistWebsite(url) {
+    if (this.autoBlacklist) {
+      chrome.storage.local.get(['blacklist', 'activationFrequency'], (data) => {
+        let blacklistURLs = data.blacklist.slice(1, -1).split('|'),
+          activationFrequency = JSON.parse(data.activationFrequency),
```

Comment:
I suggest renaming this to "activationCount". Frequency is the rate of something over time. But here you are not taking time into account. Therefore, the name "frequency" is misleading.